### PR TITLE
User feed page

### PR DESCRIPTION
**Pull request by yashrajoria3**
create frontend for user feed

### CODE REVIEW

File: src/app/_components/myfeed/LeftSideBar.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+import BookmarkBorderOutlinedIcon from "@mui/icons-material/BookmarkBorderOutlined";
+import ShoppingCartOutlinedIcon from "@mui/icons-material/ShoppingCartOutlined";
+import SettingsOutlinedIcon from "@mui/icons-material/SettingsOutlined";
+import AccountCircleOutlinedIcon from "@mui/icons-material/AccountCircleOutlined";
+import userImage from "~/assets/images/user.jpeg";
+import Image from "next/image";
+import { useRouter } from "next/navigation";
+import { type MenuItem } from "~/app/myfeed/types/types";
+
+interface Props {
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/LeftSideBar.tsx]: Having appropriate names are important. Name it LeftSidebarProps

File: src/app/_components/myfeed/LeftSideBar.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+          ></Image>
+          <Typography
+            sx={{
+              fontSize: "17px",
+              fontWeight: "semibold",
+              color: "text.secondary",
+              marginLeft: "4px",
+            }}
+          >
+            Steve Jobs
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/LeftSideBar.tsx]: We need to show the logged in user name here

File: src/app/_components/myfeed/RightSideBar.tsx [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+import React from "react";
+import FavoriteBorderIcon from "@mui/icons-material/FavoriteBorder";
+import Link from "next/link";
+import {
+  type FeaturedArticle,
+  type WriterToFollow,
+} from "~/app/myfeed/types/types";
+import { followWriter } from "~/app/myfeed/types/types";
+
+interface Props {
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/RightSideBar.tsx]: call it RightSidebarProps

File: src/app/myfeed/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+          py: "4px",
+        }}
+      >
+        <Grid
+          size={2}
+          sx={{
+            height: "90vh",
+          }}
+        >
+          <LeftSideBar menuItems={MENU_ITEMS} />
```
Comment (thekalamanchofficial) [inline src/app/myfeed/layout.tsx]: You added this to myfeed layout, Will it show for other routes as well?

File: src/app/myfeed/static/static.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
+const featuredArticles = [
```
Comment (thekalamanchofficial) [inline src/app/myfeed/static/static.ts]: this is actually mock data. call the folder myFeedMock and file myFeed.mock.ts

File: src/app/myfeed/static/static.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    profilePicture: "https://picsum.photos/200",
+    writerName: "Steve Jobs",
+    articleName: "सपनों के परिंदे",
+    articleLink: "/",
+    writerProfileLink: "/",
+    likes: 320,
+  },
+];
+
+const writersToFollow = [
```
Comment (thekalamanchofficial) [inline src/app/myfeed/static/static.ts]: Always name these type constants like this WRITERS_TO_FOLLOW_MOCK

File: src/app/myfeed/types/types.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  writerProfileLink: string;
+}
+
+export interface MenuItem {
+  label: string;
+  route: string;
+  icon: React.ReactNode;
+}
+
+// import { currentUser } from "@clerk/nextjs/server";
```
Comment (thekalamanchofficial) [inline src/app/myfeed/types/types.ts]: remove commented code

File: src/theme.ts [hunk 8]
```diff
@@ -1,6 +1,20 @@
 "use client";
 import { createTheme } from "@mui/material/styles";
 
+declare module "@mui/material/styles" {
```
Comment (thekalamanchofficial) [inline src/theme.ts:4]: why did we have to do declare module here? This should be there by default

File: src/app/_components/myfeed/PostsFeed.tsx [hunk 9]
```diff
@@ -0,0 +1,869 @@
...
+import React from "react";
+import userImage from "~/assets/images/user.jpeg";
+import MoreHorizOutlinedIcon from "@mui/icons-material/MoreHorizOutlined";
+import FavoriteBorderIcon from "@mui/icons-material/FavoriteBorder";
+import MessageIcon from "@mui/icons-material/Message";
+import SendIcon from "@mui/icons-material/Send";
+import TollIcon from "@mui/icons-material/Toll";
+import BookmarkBorderIcon from "@mui/icons-material/BookmarkBorder";
+
+const PostsFeed = () => {
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/PostsFeed.tsx]: The component is still seems to be hardcoded. Declare correct props for this. Break out the reusable component such as  Profile and follow buttons, like, comment and other buttons. Declare appropriate props for them as well

File: next.config.js [hunk 10]
```diff
@@ -6,6 +6,18 @@ await import("./src/env.js");
 
 /** @type {import('next').NextConfig} */
 const config = {
+  images: {
+    domains: ["picsum.photos"], // Add the hostname here
```
Comment (thekalamanchofficial) [inline next.config.js]: What's this?

File: next.config.js [hunk 11]
```diff
@@ -6,6 +6,18 @@ await import("./src/env.js");
...
+  },
+  async redirects() {
```
Comment (thekalamanchofficial) [inline next.config.js:19]: what's this?

File: src/app/_components/myfeed/PostsFeed.tsx [hunk 12]
```diff
@@ -0,0 +1,869 @@
...
+                Repellendus quaerat, delectus in modi cumque unde sit nam vitae
+                assumenda enim pariatur, suscipit facere explicabo est deserunt
+                obcaecati accusantium commodi sapiente! asdasd
+                <Button
+                  variant="text"
+                  sx={{
+                    minHeight: "auto",
+                    width: "auto",
+                    height: "10px",
+                    // display: "inline",
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/PostsFeed.tsx]: remove the commented codes

File: src/app/_components/myfeed/PostsFeed.tsx [hunk 13]
```diff
@@ -0,0 +1,869 @@
...
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/PostsFeed.tsx]: remove commented code

File: src/app/_components/myfeed/RightSideBar.tsx [hunk 14]
```diff
@@ -0,0 +1,286 @@
...
+          px: "12px",
+          py: "8px",
+        }}
+      >
+        <Typography
+          sx={{
+            color: "text.primary",
+            fontSize: "18px",
+            fontWeight: "550",
+            // py: "5px",
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/RightSideBar.tsx:143]: remove commented code

File: src/app/layout.tsx [hunk 15]
```diff
@@ -1,19 +1,13 @@
...
 import theme from "../theme";
 
-const inter = Inter({
-  subsets: ["latin"],
-  display: "swap",
-});
-
 export const metadata: Metadata = {
   title: "Kalamanach",
   description: "A Platform for Writers and Poets",
```
Comment (thekalamanchofficial) [inline src/app/layout.tsx]: Call it `A platform for artists`

File: src/app/myfeed/page.tsx [hunk 16]
```diff
@@ -0,0 +1,78 @@
+"use client";
+import { Grid2 as Grid, Tab, Tabs } from "@mui/material";
+import React, { useState } from "react";
+import PostsFeed from "../_components/myfeed/PostsFeed";
+const MyFeed = () => {
+  const [value, setValue] = useState(0);
```
Comment (thekalamanchofficial) [inline src/app/myfeed/page.tsx]: Can't we use enums instead of using index for each tab

**Review by thekalamanchofficial — CHANGES_REQUESTED** [review]

File: next.config.js (hunk 11 above)
Comment (yashrajoria3) [reply next.config.js:19]: to redirect from / to /myfeed
basically to set /myfeed as default

File: src/app/myfeed/layout.tsx (hunk 4 above)
Comment (yashrajoria3) [reply src/app/myfeed/layout.tsx]: when i added this in the global one, it was showing for others as well for which it was not needed 
but if we add this in my feed folder all the routes that starts with /myfeed will get this layout
need to check this again

File: src/app/_components/myfeed/LeftSideBar.tsx (hunk 2 above)
Comment (yashrajoria3) [reply src/app/_components/myfeed/LeftSideBar.tsx]: there was some issue with this, it was saying we need a server side component for this 
and we have all the client side component for now

File: next.config.js (hunk 10 above)
Comment (yashrajoria3) [reply next.config.js]: to allow dummy images from picsum website

File: src/app/myfeed/page.tsx (hunk 16 above)
Comment (yashrajoria3) [reply src/app/myfeed/page.tsx]: it seems to be working for integers only

File: src/app/_components/myfeed/RightSideBar.tsx [hunk 17]
```diff
@@ -0,0 +1,259 @@
...
+import React from "react";
+import FavoriteBorderIcon from "@mui/icons-material/FavoriteBorder";
+import Link from "next/link";
+import { followWriter } from "~/app/myfeed/types/types";
+import UserNameProfile from "./UserNameProfile";
+
+import { type RightSideBarProps } from "~/app/myfeed/types/types";
+
+const RightSideBar: React.FC<RightSideBarProps> = ({
+  featuredArticles: featuredArticles,
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/RightSideBar.tsx]: use destructured

File: src/app/_components/myfeed/RightSideBar.tsx [hunk 18]
```diff
@@ -0,0 +1,259 @@
...
+                      style={{ textDecoration: "none" }}
+                    >
+                      <UserNameProfile
+                        ImageHeight={25}
+                        ImageWidth={25}
+                        NameFontSize={15}
+                        NameFontWeight="400"
+                      />
+                    </Link>
+                    <Box
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/RightSideBar.tsx:102]: create components for all like, comment, bids, forward buttons and use them

File: src/app/_components/myfeed/RightSideBar.tsx [hunk 19]
```diff
@@ -0,0 +1,259 @@
...
+                <Box
+                  sx={{
+                    width: "100%",
+                    display: "flex",
+                    flexDirection: "row",
+                    justifyContent: "start",
+                    alignItems: "start",
+                    pt: "3px",
+                  }}
+                >
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/RightSideBar.tsx:163]: move to a component, call it WritersToFollowButton

File: src/app/myfeed/myfeedMock/myfeedMock.ts [hunk 20]
```diff
@@ -0,0 +1,148 @@
...
+    writerName: "Steve Jobs",
+    articleName: "सपनों के परिंदे",
+    articleLink: "/",
+    writerProfileLink: "/",
+    likes: 320,
+  },
+];
+
+const WRITERS_TO_FOLLOW_MOCK = [
+  {
```
Comment (thekalamanchofficial) [inline src/app/myfeed/myfeedMock/myfeedMock.ts]: [100].map(()=>{})

File: src/app/myfeed/types/types.ts [hunk 21]
```diff
@@ -0,0 +1,65 @@
+export const followWriter = (writerName: string, writerProfileLink: string) => {
+  console.log(`followed writer ${writerName} ${writerProfileLink} `);
+};
+
+interface FeaturedArticle {
```
Comment (thekalamanchofficial) [inline src/app/myfeed/types/types.ts]: change all interface to types

File: src/app/_components/myfeed/PostCardContent.tsx [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+          {articleTitle}
+        </Typography>
+        <Typography
+          sx={{
+            color: "text.secondary",
+            fontSize: "15px",
+            marginBottom: "10px",
+          }}
+        >
+          {articleContent.slice(0, 400)}
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/PostCardContent.tsx]: store the constant in config

File: src/app/_components/myfeed/PostCardContent.tsx [hunk 23]
```diff
@@ -0,0 +1,161 @@
...
+          {articleContent.length > 400 ? " ..." : ""}
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/PostCardContent.tsx]: use one ternary operator to all three expressions

File: src/app/_components/myfeed/PostCardContent.tsx [hunk 24]
```diff
@@ -0,0 +1,161 @@
...
+
+          {articleContent.length > 400 && (
+            <Button
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/PostCardContent.tsx]: use the custom see more button here

File: src/app/_components/myfeed/PostCardContent.tsx [hunk 25]
```diff
@@ -0,0 +1,161 @@
...
+            {articleTitle}
+          </Typography>
+          <Typography
+            sx={{
+              color: "text.secondary",
+              fontSize: "15px",
+              marginBottom: "10px",
+            }}
+          >
+            {articleDescription.slice(0, 250)}
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/PostCardContent.tsx]: use single expression to render this and use custom see more button

File: src/app/_components/myfeed/PostCardFooter.tsx [hunk 26]
```diff
@@ -0,0 +1,86 @@
...
+    >
+      <Box
+        sx={{
+          display: "flex",
+          justifyContent: "start",
+          alignItems: "center",
+          gap: "10px",
+        }}
+      >
+        <Chip
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/PostCardFooter.tsx]: use the custom chip buttons created here. Add the callbacks in the custom buttons created

File: src/app/_components/myfeed/PostsFeed.tsx [hunk 27]
```diff
@@ -0,0 +1,83 @@
...
+                      backgroundColor: "secondary.main",
+                      color: "primary.main",
+                      minHeight: "auto",
+                      height: "40px",
+                      padding: "2px 10px ",
+                      margin: "3px 5px ",
+                      fontSize: "15px",
+                    }}
+                  >
+                    Follow
```
Comment (thekalamanchofficial) [inline src/app/_components/myfeed/PostsFeed.tsx]: create a custom Follow button

**Comment by yashrajoria3** [comment]
implemented the changes

**Review by thekalamanchofficial — APPROVED** [review]

**Review by helloManojkr — APPROVED** [review]

**Pull request merged**